open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +2991,76 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot
+          type: integer
+          format: int64
+        snapshot-range:
+          description: A JSON list containing exactly 2 snapshot IDs representing the start (exclusive) and end (inclusive) snapshots. This option is not allowed when `snapshot-id` is present in the request.
+          type: array
+          items:
+            type: integer
+            format: int64
+      oneOf:
+        - required: [snapshot-id]
```

Comment:
to what does the `oneOf` apply? This also doesn't properly get translated in https://editor.swagger.io/

Reply:
I think your right, i basically wanted to express that user can only set `snapshot-id` or `snapshot-range` but not both. 

I found this https://stackoverflow.com/questions/21134029/how-to-define-mutually-exclusive-query-parameters-in-swagger-openapi but it seems that mutually exclusive properties in request body is not possible and that we may have to just rely on the descriptions of the fields to tell the user to only pass one of these but not both.